util/dogfood.sh

```diff
@@ -0,0 +1 @@
+rm -rf target* && cargo build --lib && cp -R target target_recur && cargo rustc -- -Zextra-plugins=clippy -Ltarget_recur/debug   -Dclippy
```

Comment:
Yes, and while it's fine for Travis to not remove anything, running dogfood.sh locally needs you to clean the target dir (older artefacts accumulate).

I modified it so that it doesn't delete everything, just the final so files